### ISSUE TITLE
Server is crashing when map cycle changing map to the next

### ISSUE THREAD

**Issue by paulogagliardi** [fte-team/fteqw#220]
Please...  the server is crashing when map cycle changing map to the next, using mapcycle.qc. 

--------- Initializing MapCycle ----------
--------- Initializing MapCycle ----------
mapcycle initialized with 346 entries.
Server spawned.
execing senha.cfg
couldn't exec entities/misdm5.ent
COM_WriteFile: ssqccore.txt
[0;1;31;40mSV_Error: PF_readcmd: map changed before reading[0;m

Server ended

Fatal error: SV_Error: PF_readcmd: map changed before reading

Error: signal SIGABRT:
/lib/x86_64-linux-gnu/libc.so.6(gsignal+0xcb)[0x7f63e19b400b]
/lib/x86_64-linux-gnu/libc.so.6(abort+0x12b)[0x7f63e1993859]
./fteqw-sv(+0x1e1388)[0x5640d7699388]
./fteqw-sv(+0x121702)[0x5640d75d9702]
./fteqw-sv(+0x12189a)[0x5640d75d989a]
./fteqw-sv(+0x109b20)[0x5640d75c1b20]
./fteqw-sv(+0x1ddae8)[0x5640d7695ae8]
./fteqw-sv(+0x1dff2e)[0x5640d7697f2e]

Not certain, but I think the server is loading next map, before ending previous.

A good solution could be a break of 10 seconds after the end of a map and the beginning of the next?

**Comment by eukara** [comment]
Just got back from my weekend.
Sounds like some sort of race condition or an issue with your progs.dat if you've customized it. I've been running the server for a while now without any such issue. I will look into avoid using readcmd and add some more safety checks around when/how the data is read and will update here when that's done.

**Comment by paulogagliardi** [comment]
Hhmmm... in fact this error is really only in Total Destruction mod. 
Well... I already had removed the old references to maps in code before the implementation of mapcycles.qc. (before mapcycle.qc TD rotated only original quake maps... and it was boring)

The code is here: https://github.com/paulogagliardi/td2qw 

The old references to maps was in the status.qc and strings.qc. Now, I am not sure that I remove all references.

**Comment by Shpoike** [comment]
readcmd is an mvdsv builtin that I really don't want to have to support properly... the fact that there's deprecation warnings on it inside fteextensions.qc should be a strong enough hint to never use it...
If you're using it to execute the 'map' command or some other related console command then you'll have destroyed all the qc state - without that check, the engine would basically end up executing gibberish, hence why it aborts (and is unable to print any qc stack trace).
the engine's stack trace probably isn't very useful to you - dedicated servers crash out where listen servers would simply crash to console instead.

chances are there's a map command still sitting in the console buffer that needs to be execed first to ensure that console command order is properly obeyed, otherwise it'd be unable to handle aliases properly etc.

that code isn't even using the returned string, so just swap them out for the much safer localcmd builtin.

**Comment by paulogagliardi** [comment]
I'm not sure if it has anything to do with the question, but:

I changed the FTEQW version and the error, strangely, stopped occurring.
I'm still not sure this resolved the issue. But in fact I haven't received the error anymore. Playing against other players and the server is no longer crashing.

My previous version of FTEQW was a version compiled on the server itself (64xx if I'm not mistaken). So I downloaded version 6202 from https://fte.triptohell.info/.

Still monitoring.
